### ISSUE TITLE
Register and support opening rdp files directly

### ISSUE THREAD

**Issue by freechelmi** [FreeRDP/Remmina#1105]
Nice that we can now drag n drop rdp files to open them.

this issue it to be able to open files as a default file handler for rdp files .

This means "open with - remmina " from a file explorer would connect directly to the rdp server.

* Client (OS name and version): any 
* Remmina version (remmina --version):  1.2.0-rcgit-17

**Comment by antenore** [comment]
Yes, this is something we were looking for, but honestly we didn't find yet how to do it, even if I'm sure is something really easy to be implemented.

Any help is welcome.

**Comment by MagicFab** [comment]
I can see how this would be useful, perhaps even more to import existing RDP files.

I couldn't find information about how RDP files are formatted, only these two links:

- https://engrmosaic.uncc.edu/mosaic-anywhere/remote-desktop-file-format
- https://msdn.microsoft.com/en-us/library/gg156129(v=winembedded.70).aspx

**Comment by antenore** [comment]
@MagicFab , yes are the only sources I've found as well.

Remmina has almost everything is needed except:

1. Register a Mime-type for .remmina files (while we are on it ;-) )
2. Register a Mime-type for .rdp files
3. Use the existing import remmina functions to read and use the .rdp passed as argument.

Hope it helps, unlucky at the moment I'm too busy with other tasks, so I cannot work on it.

**Comment by antenore** [comment]
@BloodyIron can you please close remmina and execute it from a terminal, than copy and paste here the output?

**Comment by BloodyIron** [comment]
Should I just run it "remmina server.rdp" or what?

**Comment by antenore** [comment]
As you do usually, but execute remmina from the command line before to try. Executing remmina from the command line will print to the standard output some logs that may be useful for me.

To do so be sure to have closed all the remmina sessions, for example from the command line 

```
pkill remmina ; remmina
```

**Comment by BloodyIron** [comment]
The only output produced when trying to "run" the server.rdp is:

(remmina:17332): Gtk-WARNING **: gtk_menu_attach_to_widget(): menu already attached to GtkMenuItem

And it still exhibits the behaviour of opening the remmina server selection window, but does not ask for connection credentials (like it was doing when it was working).

**Comment by antenore** [comment]
which version of Remmina do you have and on which Liux distribution/version?

**Comment by BloodyIron** [comment]
Remmina: 1.2.0-rcgit-27
I'm using this PPA : http://ppa.launchpad.net/remmina-ppa-team/remmina-next/ubuntu

Ubuntu 17.10 + XFCE4 + Compiz
Linux 4.13.0-36-generic

Any more useful info I can pass to you? :^)

**Comment by antenore** [comment]
thanks, that's all for today, here is bed time now :-) 
Tomorrow I'll test it.

Just one more note, I usually drag&drop an rdp file in the remmina main window to import it or the import/export shortcut in the main menu.

**Comment by BloodyIron** [comment]
The RDP stuff I'm doing keeps changing, hence why in this case I was liking it just connecting without importing. Drag and Drop does work for importing the RDP config, but the public IPs for the stuff I'm working against keeps changing (which is intentional), so that's a pig to deal with :^)

**Comment by antenore** [comment]
@BloodyIron running

```remmina -c /path/to/file.rdp ```

Works for me. Can you try?

**Comment by BloodyIron** [comment]
@antenore that appears to launch the rdp connection as one would want.

**Comment by antenore** [comment]
Yes, drag and drop acts only as import.

**Comment by BloodyIron** [comment]
I was not referring to drag and drop, I was meaning, when I download the rdp config file from (Azure in this case), and tell my browser to "open" it with remmina. Earlier it was starting the connection, and prompting (accept cert? login creds). But recently it just opens the Remmina GUI (showing profiles), and does nothing.

I would like Remmina to "run" the rdp files again, as that is quite convenient for me at times. :^)

**Comment by antenore** [comment]
May I have an unonymized copy of the rdp file please?

**Comment by BloodyIron** [comment]
Is there a way I can pass this to you privately?

**Comment by BloodyIron** [comment]
I just emailed you two. Let me know if you didn't get them.

**Comment by antenore** [comment]
Got it thanks.

I'll need sometime to tests and understand what happens behind, I'll come back to you by email and than, when we will be done, I'll post here "the answer". :-)

**Comment by BloodyIron** [comment]
Excellent. As a heads-up, the IP used for those machines will change next time they shut down, which is probably later today (auto-shutdown at 7pm my time). So with any luck that won't complicate this for you, sorry!

**Comment by antenore** [comment]
np. I won't connect at all, I just need to understand what's going on. Honestly I've never tried to open a connection directly from an rdp file.

**Comment by antenore** [comment]
So, as far as I remember this has never been support in the Remmina 1.2 serie, probably in the previous one? We don't maintain that jurassic code since 3 years now, so I confirm that this is not a bug but a feature request :-) 

I'll try to implement for the next release

**Comment by antenore** [comment]
I'm sorry, I'm bombing you with tons of questions.
Can you test with xfreerdp from the command line and see if you get the same errors?

Executing just xfreerdp from the command line will print the help.
Something like this should suffice:

```sh
xfreerdp +clipboard  /v:hostname /u:username
```
Replacing user and hostname.

If you have the same error, unlucky, you have to open a bug to the FreeRDP team, if they will fix it quickly, I can release a new FreeRDP package of Remmina.

https://github.com/FreeRDP/FreeRDP/issues/

**Comment by antenore** [comment]
Wrong issue sorry, deleting the post

**Comment by BloodyIron** [comment]
@antenore well, be that as it may, somehow the behaviour exhibited without any remmina version change though. So... it's quite peculiar. But if we could get it baked in, that would be really appreciated! :D

**Comment by antenore** [comment]
Implemented via #1497

**Comment by BloodyIron** [comment]
@antenore woot! :D Thanks!
